packages/flutter/test/foundation/leak_tracking.dart

```diff
@@ -5,8 +5,115 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:leak_tracker/leak_tracker.dart';
+import 'package:meta/meta.dart';
 
-typedef LeaksObtainer = void Function(Leaks foundLeaks);
+/// Set of objects, that does not hold the objects from garbafge collection.
+///
+/// The objects are referenced by hash codes and can duplicate with low probability.
+@visibleForTesting
+class WeakSet {
+  final Set<String> _objectCodes = <String>{};
+
+  String _toCode(int hashCode, String type) => '$type-$hashCode';
+
+  void add(Object object) {
+    _objectCodes.add(_toCode(identityHashCode(object), object.runtimeType.toString()));
+  }
+
+  void addByCode(int hashCode, String type) {
+    _objectCodes.add(_toCode(hashCode, type));
+  }
+
+  bool contains(int hashCode, String type) {
+    final bool result = _objectCodes.contains(_toCode(hashCode, type));
+    return result;
+  }
+}
+
+@visibleForTesting
+class TestWidgetsConfig {
+  final WeakSet heldObjects = WeakSet();
+}
+
+/// Extension for [WidgetTester] to configure held objects.
```

Comment:
Do we still need this with our current decision to not track non-gced leaks? Currently, it doesn't matter whether you mark something as held object or not, right? If so, let's remove this for now to avoid that it is used incorrectly until it actually matters.

Reply:
Looks like this is also unused in this PR?